Add condition for Satellite specific step (#1632)
theforeman/foreman-documentation: guides/common/modules/proc_installing-openscap-plugin.adoc

+
[options="nowrap" subs="+quotes,attributes"]
----

> # {foreman-installer} --enable-foreman-plugin-openscap  --enable-foreman-proxy-plugin-openscap

Reviewer: ```suggestion
# {foreman-installer} --enable-foreman-plugin-openscap --enable-foreman-proxy-plugin-openscap
```

----
ifndef::satellite[]
. Optional: Install the OpenSCAP Hammer CLI plug-in on your {ProjectServer}:
+
[options="nowrap" subs="+quotes,attributes"]
----
# {foreman-installer} --enable-foreman-cli-openscap
----
endif::[]

> 

Reviewer 2: ```suggestion
```

> . Install the OpenSCAP plug-in on any {SmartProxies}:

Reviewer 2: ```suggestion
. Install the OpenSCAP plug-in on any {SmartProxyServer}s:
```

+
[options="nowrap" subs="+quotes,attributes"]
----
# {foreman-installer} --enable-foreman-plugin-openscap  --enable-foreman-proxy-plugin-openscap
----
ifndef::satellite[]

> . Optional: Add `--enable-foreman-cli-openscap` for {ProjectServer} above.

Reviewer 2: ```suggestion
. Optional: Install the OpenSCAP Hammer CLI plug-in on your {ProjectServer}:
+
[options="nowrap" subs="+quotes,attributes"]
----
# {foreman-installer} --enable-foreman-cli-openscap
----
```

+
[options="nowrap" subs="+quotes,attributes"]
----
# {foreman-installer} --enable-foreman-plugin-openscap  --enable-foreman-proxy-plugin-openscap
----
ifndef::satellite[]
. Optional: Install the OpenSCAP Hammer CLI plug-in on your {ProjectServer}:
+
[options="nowrap" subs="+quotes,attributes"]
----
# {foreman-installer} --enable-foreman-cli-openscap
----
endif::[]

. Install the OpenSCAP plug-in on any {SmartProxies}:
+
[options="nowrap" subs="+quotes,attributes"]
----
# {foreman-installer} --enable-foreman-proxy-plugin-openscap
----
ifndef::satellite[]
. Optional: Install the OpenSCAP Hammer CLI plug-in:
+
[options="nowrap" subs="+quotes,attributes"]
----
# {foreman-installer} --enable-foreman-cli-openscap
----

> endif::[]

Reviewer: Please drop this (line 29 - 36). Installing the CLI is invalid on Smart Proxies and it's already documented above for the main server.

----
+

> Perform this command on both your {ProjectServer} and any attached {SmartProxies}.

Author: @ekohl gentle reminder!

> Perform this command on both your {ProjectServer} and any attached {SmartProxies}.

Reviewer: That's indeed what I meant.

> Perform this command on both your {ProjectServer} and any attached {SmartProxies}.

Reviewer: Though I really have difficulty figuring out what "this command" is referring to. Is it referring to the previous (`--enable-foreman-proxy-plugin-openscap`) or the next (`--enable-foreman-cli-openscap`). If it's referring to the previous, should it actually be said before the command?

@maximiliankolb any thoughts on this?

> Perform this command on both your {ProjectServer} and any attached {SmartProxies}.

Reviewer 2: IMHO "this" is part of the command above:

````diff
diff --git a/guides/common/modules/proc_installing-openscap-plugin.adoc b/guides/common/modules/proc_installing-openscap-plugin.adoc
index e7c677a9c..b4b7212a3 100644
--- a/guides/common/modules/proc_installing-openscap-plugin.adoc
+++ b/guides/common/modules/proc_installing-openscap-plugin.adoc
@@ -11,14 +11,12 @@ The OpenSCAP plug-in consists of the main OpenSCAP plug-in itself, the OpenSCAP
 ----
 # {foreman-installer} --enable-foreman-plugin-openscap
 ----
-. Install the OpenSCAP plug-in on your {SmartProxies}:
+. Install the OpenSCAP plug-in on both your {ProjectServer} and any attached {SmartProxies}:
 +
 [options="nowrap" subs="+quotes,attributes"]
 ----
 # {foreman-installer} --enable-foreman-proxy-plugin-openscap
 ----
-+
-Perform this command on both your {ProjectServer} and any attached {SmartProxies}.
 . Optional: Install the OpenSCAP Hammer CLI plug-in:
 +
 [options="nowrap" subs="+quotes,attributes"]
````

> Perform this command on both your {ProjectServer} and any attached {SmartProxies}.

Reviewer: That's also how I read it and I think your suggestion improves it. I'd like to mention that technically that they only need to run it on Smart Proxies where they want to use OpenSCAP on clients, but that's a bit of an edge case.

What we can also do is (roughly):

* Run this on {ProjectServer}
{foreman-installer} --enable-foreman-plugin-openscap --enable-foreman-proxy-plugin-openscap
* Run this on any {SmartProxies}
{foreman-installer} --enable-foreman-proxy-plugin-openscap

That saves the user an installer run on {ProjectServer} and thus the procedure is quicker and also easier to follow.

It should also be noted that `--enable-foreman-cli-openscap` only works on the `katello` and `foreman` scenarios (and their downstream equivalents), but is invalid for the `foreman-proxy-content` scenario.

That suggests a rough flow like:

* Run this on {ProjectServer}
{foreman-installer} --enable-foreman-plugin-openscap --enable-foreman-proxy-plugin-openscap
IF NOT SATELLITE
* If desired, add `--enable-foreman-cli-openscap` to the above command
ENDIF
* Run this on any {SmartProxies}
{foreman-installer} --enable-foreman-proxy-plugin-openscap

To emphasize, I'm only trying to describe the structure of the instructions, not the phrasing.

Does that make sense? Other suggestions? I realize it increases the scope of the issue, but I think it's a good chance to improve things.

> Perform this command on both your {ProjectServer} and any attached {SmartProxies}.

Author: @ekohl 

Considering this, line 7 to 21 in the current document:

```
.Procedure
. Install the OpenSCAP plug-in on your {ProjectServer}:
+
[options="nowrap" subs="+quotes,attributes"]
----
# {foreman-installer} --enable-foreman-plugin-openscap
----
. Install the OpenSCAP plug-in on your {SmartProxies}:
+
[options="nowrap" subs="+quotes,attributes"]
----
# {foreman-installer} --enable-foreman-proxy-plugin-openscap
----
+
Perform this command on both your {ProjectServer} and any attached {SmartProxies}.
```
Will be replaced by the following:

```
.Procedure
. Install the OpenSCAP plug-in on your {ProjectServer}:
[options="nowrap" subs="+quotes,attributes"]
----
# {foreman-installer} --enable-foreman-plugin-openscap  {foreman-installer} --enable-foreman-proxy-plugin-openscap
----
ifndef::satellite[]
If required, add --enable-foreman-cli-openscap for {ProjectServer}
endif::[]

. Install the OpenSCAP plug-in on any {SmartProxies}:
+
[options="nowrap" subs="+quotes,attributes"]
----
# {foreman-installer} --enable-foreman-proxy-plugin-openscap
----
```

> Perform this command on both your {ProjectServer} and any attached {SmartProxies}.

Reviewer: > # {foreman-installer} --enable-foreman-plugin-openscap  {foreman-installer} --enable-foreman-proxy-plugin-openscap

You have repeated `{foreman-installer}`. It should be
```
# {foreman-installer} --enable-foreman-plugin-openscap --enable-foreman-proxy-plugin-openscap
```

> If required, add --enable-foreman-cli-openscap for {ProjectServer}

Perhaps this should be explicitly mentioning that it's `Optional:` and that it's for the above command?

> Perform this command on both your {ProjectServer} and any attached {SmartProxies}.

Author: @ekohl 
Corrected in the same box above.

> ifdef::foreman-el,katello[]

Reviewer 3: Not sure about Orcharhino, @maximiliankolb ?

. Optional: Install the OpenSCAP Hammer CLI plug-in:
+
[options="nowrap" subs="+quotes,attributes"]
----
# {foreman-installer} --enable-foreman-cli-openscap
----
endif::[]

> . Install the OpenSCAP plug-in Puppet module:

Reviewer: To simplify this in future releases I opened https://github.com/theforeman/puppet-foreman_proxy/pull/781 so it's handled by the installer.

Author: @ekohl 
I am not sure, what action item for this is expected from me. All the other things are fixed in this PR except this.

Reviewer: With Foreman 3.5 you'll be able to pass `--foreman-proxy-plugin-openscap-puppet-module true` instead of a manual package installation. So in the short term nothing, but a follow up PR should be created to use that.

Author: ok. So resolving it for now.
